shotover-proxy/src/transforms/redis/sink_cluster.rs

```diff
@@ -859,27 +856,21 @@ async fn receive_frame_response(receiver: oneshot::Receiver<Response>) -> Result
     }
 }
 
-#[inline(always)]
-fn send_frame_response(
-    one_tx: oneshot::Sender<Response>,
-    frame: RedisFrame,
-) -> Result<(), Response> {
-    one_tx.send(Response {
-        original: Message::from_frame(Frame::None),
-        response: Ok(vec![Message::from_frame(Frame::Redis(frame))]),
-    })
-}
-
-fn immediate_responder() -> (
-    oneshot::Sender<Response>,
-    impl Future<Output = Result<Response>>,
-) {
+fn short_circuit(frame: RedisFrame) -> Result<ResponseFuture> {
     let (one_tx, one_rx) = oneshot::channel::<Response>();
-    (one_tx, async {
+
+    one_tx
+        .send(Response {
+            original: Message::from_frame(Frame::None),
+            response: Ok(vec![Message::from_frame(Frame::Redis(frame))]),
```

Comment:
Do we ever put more than one message in this vec? I couldn't find any case of it.

Reply:
Very nice find!
I raised a separate PR https://github.com/shotover/shotover-proxy/pull/660/files